utils/models.py

```diff
@@ -8,3 +8,122 @@ class UUIDModel(models.Model):
 
     class Meta:
         abstract = True
+
+
+class SerializableMixin(models.Model):
+    """
+    Mixin to add custom serialization for django models in order to get the desired tree of models to
+    the downloadable JSON form. It detects relationships automatically (many to many not yet fully
+    supported). Check for the example for more details about the structure.
+
+    Attributes need to be defined in the extending model:
+
+    - serialize_fields (required)
+        - tuple of dicts:
+            - name (required), name of the field or relation that's going to be added to the serialized object
+            - accessor (optional), function that is called when value of the field is resolved and it takes the
+              actual field value as argument
+
+    - serialize_root_key (required only for the root object)
+        - it's used as key ONLY for the root object (where serializing starts)
```

Comment:
is this serialize_root_key just for some verbosity? don't understand the technical value behind it

well, `serialize()` will return a list of the serialized objects and not a dict, but it will still serialize

Reply:
Excellent point there 🎖. Originally it ended up there basically just because the spec said that `Profile` node needs to have a key with value "CUSTOMER" instead of "PROFILE". I think if we wan't to rename the root key we can do it in the resolver when we are about to resolve the JSON.

I removed the `serialize_root_key`completely and I think it led to a cleaner and more generic solution. I refactored `serialize()` to always return:

```
{
  "key": {model_name_here}
  "children": {fields_here}
}
```

This way now I can just return a list of `obj.serialize()` in manager's serialize method. Only thing I needed to change besides above was to make 1-to-1 relation an exception in order to avoid this

```
{
  "key": "SENSITIVEDATA"
  "children": [
    {
      "key": "SENSITIVEDATA"
      "children": [
        ...
      ]
    }
  ]
}
```

Reply:
yeah, looks simpler now 👍